src/Symfony/Component/Form/Form.php

```diff
@@ -532,6 +532,9 @@ public function submit($submittedData, $clearMissing = true)
             $submittedData = null;
         } elseif (is_scalar($submittedData)) {
             $submittedData = (string) $submittedData;
+        } elseif (\is_array($submittedData) && !$this->config->getCompound() && !$this->config->hasOption('multiple')) {
```

Comment:
is https://github.com/symfony/symfony/pull/29307/files#diff-ca5e25b47f3ecc94cd557946aeb486c6R567 still needed?

Reply:
yes it is: eg events can alter the submitted data and return something invalid that this would catch.